### PR TITLE
Documentation for KMeterISO component

### PR DESCRIPTION
**Pull request by Rudd-O**
## Description:

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/2333

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5170

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  **Please disregard the name of the origin branch.  It's `next` even though it's named `current`.**
 
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by jesserockz** [comment]
Please either recreate your PR with a new branch from `next` and do not use `next` or `current` directly for PR changes

**Comment by Rudd-O** [comment]
I've force-pushed to my "current" branch in my repo, but it's really the "next" branch (it's just a name issue).  You'll see five independent commits without any other extraneous junk now.

File: components/sensor/kmeteriso.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+.. code-block:: yaml
+
+    # Example configuration entry
+    i2c:
+      sda: GPIO21
+      scl: GPIO22
+      scan: true
+      id: bus_a
+      frequency: 100kHz
+
```
Comment (jesserockz) [inline components/sensor/kmeteriso.rst]: This is not required in the example
```suggestion
```

File: components/sensor/kmeteriso.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    sensor:
+    - platform: kmeteriso
+      temperature:
+        name: Temperature
+      internal_temperature:
+        name: Internal temperature
+        entity_category: diagnostic
+      update_interval: 10s
```
Comment (jesserockz) [inline components/sensor/kmeteriso.rst]:
```suggestion
```

File: components/sensor/kmeteriso.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  - All other options from :ref:`Sensor <config-sensor>`.
+
+- **internal_temperature** (*Optional*): The information for the temperature sensor inside the probe.
+
+  - **name** (**Required**, string): The name for the temperature sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - All other options from :ref:`Sensor <config-sensor>`.
+
+- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
+  sensor. Defaults to ``5s``.
```
Comment (jesserockz) [inline components/sensor/kmeteriso.rst]:
```suggestion
- **temperature** (*Optional*): The information for the temperature sensor. All options from :ref:`Sensor <config-sensor>`.
- **internal_temperature** (*Optional*): The information for the temperature sensor inside the probe. All options from :ref:`Sensor <config-sensor>`.
- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
  sensor. Defaults to ``5s``.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/sensor/kmeteriso.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+    - platform: kmeteriso
+      temperature:
+        name: Temperature
+      internal_temperature:
+        name: Internal temperature
+        entity_category: diagnostic
```
Comment (jesserockz) [inline components/sensor/kmeteriso.rst]: Sorry just noticed this one more.

If this is recommended to be diagnostic, then it should be set in the python code as diagnostic by default, and removed from here.

Comment (Rudd-O) [reply components/sensor/kmeteriso.rst]: Thanks.  How do I default this entity to diagnostic in the python code, such that the user can also choose to make it *not* diagnostic?  Thanks in advance.

Comment (Rudd-O) [reply components/sensor/kmeteriso.rst]: I think I got it.  Check latest commit here as well as https://github.com/esphome/esphome/pull/5170/commits/f3af7f909a68c10cf308286f2d77d57425e868fb

File: components/sensor/kmeteriso.rst
Comment (jesserockz) [inline components/sensor/kmeteriso.rst:1]: We dont need to add this, its just extra noise

**Review by jesserockz — APPROVED** [review]

**Comment by Rudd-O** [comment]
Thank you so much.  Please merge whenever you see fit.

**Pull request merged**